File: 通用代码/命令模式(Reciver封装)/ConcreteCommand2.java
public class ConcreteCommand2 extends Command{
    //声明自己的默认接收者
    public ConcreteCommand2(){
        super(new ConcreteReciver2());
    }
    //构造函数传递接收者
    public ConcreteCommand2(Receiver _receiver){
        super(_receiver);
    }
    //必须实现一个命令
    public void execute(){
        //业务处理
        super.receiver.doSomething();
    }
}

File: 通用代码/命令模式(Reciver封装)(C++)/ConcreteCommand2.cpp
#include "ConcreteCommand2.h"
ConcreteCommand2::ConcreteCommand2(void)
{
    //声明自己的默认接收者
    this->Command::_pReciever = new ConcreteReceiver2();
    cout << "ConcreteCommand1..." << endl;
}

ConcreteCommand2::ConcreteCommand2(Receiver* pReceiver)
{
    this->_recv = pReceiver;
    cout << "ConcreteCommand2..." << endl;
}

ConcreteCommand2::~ConcreteCommand2()
{
    cout << "~ConcreteCommand2..." << endl;
}

void ConcreteCommand2::Execute()
{
    this->_recv->doSomething();
}

